### ISSUE TITLE
Register serializer for TextTemplate earlier

### ISSUE THREAD

**Issue by Katrix** [SpongePowered/SpongeAPI#1399]
Currently the `TypeSerializer` for `TextTemplate` is registered once the class is loaded. This can cause problems if a plugin has a `TextTemplate` in a config file, and tries to load that file (thus trying to deserialize a `TextTemplate`) before the `TextTemplate` class has been loaded, because the serializer hasn't been registered yet.

https://github.com/SpongePowered/SpongeAPI/blob/bleeding/src/main/java/org/spongepowered/api/text/TextTemplate.java#L60

**Comment by Cybermaxke** [comment]
The same should be done for the Text serializer:
https://github.com/SpongePowered/SpongeAPI/blob/bleeding/src/main/java/org/spongepowered/api/text/Text.java#L88

**Comment by kashike** [comment]
https://github.com/SpongePowered/SpongeAPI/blob/bleeding/src/main/java/org/spongepowered/api/text/BookView.java#L56

**Comment by randombyte-developer** [comment]
This was a serious problem when I wanted to register my own serializer for `TextTemplate`s. My registration was to early which resulted in an overwrite by Sponge with its own serializer. To fix it for me I did `Class.forName(TextTemplate.class.getCanonicalName())` which executes the static block of `TextTemplateConfigSerializer`.

[My fix](https://github.com/randombyte-developer/kosp/blob/master/src/main/kotlin/de/randombyte/kosp/config/ConfigManager.kt#L30).

I don't know what I have destroyed with overwriting the serializer with mine. What is with other plugins that want to serialize TextTemplates, do they use my or Sponge's serializer? Wasn't there a way to locally register `TyperSerializer`s?
